### ISSUE TITLE
`steam.destkop` Exec key contains hard-coded path

### ISSUE THREAD

**Issue by fenuks** [ValveSoftware/steam-for-linux#6398]
* Steam client version (build number or date): 1.0.0.61
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Hello, provided `steam.desktop` files `Exec` command contains hard-coded binary (i.e. with full path), which is problematic if user wants to, for example, run steam in sandbox (tools like firejail create symlink in `/usr/local/bin/` directory. Could you please consider changing `Exec` key to just`steam %U`?

#### Steps for reproducing this issue:
```sh
$ sudo ln -s /usr/bin/firejail /usr/local/bin/steam
# run desktop file, exo-open is provided shipped with XFCE, other option is to use graphical launcher
$ exo-open /usr/share/applications/steam.desktop
# steam is run without sandbox because of hard-coded path in desktop file
```

**Comment by dreamer** [comment]
Just checked and, to my surprise, `steam.desktop` on Fedora (rpm package from rpmfusion) also has absolute path in `Exec` (one of very few packages to use a path in there).

**Comment by braiam** [comment]
According to [the specification](https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#exec-variables) it's allowed both ways, but in most cases this wouldn't help since /usr/local/bin usually have preferences in the $PATH (it appears first). Please check with echo $PATH.

In Debian, the steam binary is in /usr/games instead. This issue may be two steps. Arch should move steam binary to a directory later in the path and the desktop file should have the non-full path specified.

**Comment by fenuks** [comment]
You are correct that specification allows absolute paths to binary in `Exec` keys, but I think that you misunderstand reasons behind my request.

Absolute path to `/usr/bin/steam` in fact forbids me from having my shell wrapper around the steam binary put in `/usr/local/bin/steam` to do extra stuff while launching the Steam. In my case, I would like to run it in sandbox, and utility I use, firejal, creates new executable in local bin. Unfortunately, due to hard-coded path, launching Steam via desktop file results in running Steam without sandbox.

**Comment by braiam** [comment]
@fenuks I know what your request is about, I'm just informing that it's not a blocker issue. In the last paragraph I note that other lines doesn't use the full path.

**Comment by TTimo** [comment]
Hello,

Release 1.0.0.83 of the installer updates `steam.desktop` to use absolute paths more consistently. We felt this was the most appropriate way to maintain reliable functionality for users.

In order to support more advanced setups, we have detailed alternate solutions in the release notes that allow you to replace or override `steam.desktop` with a custom version: **Moderator edit:** See steam-launcher/debian/changelog in https://repo.steampowered.com/steam/archive/stable/steam_1.0.0.83.tar.gz for the previously linked details.

**Comment by fenuks** [comment]
Thank you for taking time to describe how to override default locations in changelog. ;)
